frontend/src/container/GridCardLayout/GridCardLayout.tsx

```diff
@@ -108,6 +110,27 @@ function GraphLayout({ onAddPanelHandler }: GraphLayoutProps): JSX.Element {
 		? [...ViewMenuAction, ...EditMenuAction]
 		: [...ViewMenuAction];
 
+	const handleLayoutChange = (layout: Layout[]): void => {
+		if (!isEqual(layout, dashboardLayout)) {
+			setDashboardLayout(layout);
+		}
+	};
+
+	useEffect(() => {
+		if (
```

Comment:
We should move this piece of code inside the `handleLayoutChange` function so that the `useEffect` will be removed. 

This `useEffect` will be executed every time the component mounts and will trigger an unnecessary update dashboard API call.

Reply:
It is dependent on handleLayout change event itself. however, as we dispatch event to save layout, there will be a raise condition and it doesn't update the layout with new layout

Reply:
https://www.loom.com/share/12bc0fd7da054455b5863e1ff6161cf1?sid=eb9b306a-5083-4dcd-a609-71d56148aaf5